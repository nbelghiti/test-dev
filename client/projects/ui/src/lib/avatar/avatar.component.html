<span
	class="c-avatar"
	[ngClass]="{
		'is-compact': compact
	}"
>
	<abbr [title]="name">{{initials}}</abbr>
</span>
