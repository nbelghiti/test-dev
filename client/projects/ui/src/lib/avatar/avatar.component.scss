// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Settings
// =============================================================================
$avatar-size-base: 4.4rem;
$avatar-size-compact: 3.2rem;

// =============================================================================
// :: Avatar
// =============================================================================
.c-avatar {
	display: flex;
	justify-content: center;
	align-items: center;
	width: $avatar-size-base;
	min-width: $avatar-size-base;
	height: $avatar-size-base;

	font-size: $font-size-1x-large;
	font-weight: $font-weight-bold;
	line-height: $line-height-base;
	text-transform: uppercase;
	text-align: center;

	border-radius: 50%;
	background-color: $background-color-grey-96;

	cursor: default;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-avatar {

	&.is-compact {
		width: $avatar-size-compact;
		min-width: $avatar-size-compact;
		height: $avatar-size-compact;

		font-size: $font-size-base;
	}
}
