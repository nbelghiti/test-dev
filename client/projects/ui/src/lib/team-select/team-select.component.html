<div
	class="c-team-select"
	[class.c-team-select--first]="first"
>
	<ui-heading
		[level]="3"
		[title]="title"
	>
	</ui-heading>

	<div>
		<ul class="c-team-select__members" *ngIf="members.length > 0; else emptyState">
			<li
				*ngFor="let member of members;"
				class="c-team-select__member"
			>
				<ng-container *ngTemplateOutlet="member.itemTemplate"></ng-container>
			</li>
		</ul>

		<ui-link
			*ngIf="action"
			label="New team member"
			iconBefore="plus"
			(clicked)="newMember.emit()"
		>
		</ui-link>
	</div>

	<ng-template #emptyState>
		<ui-empty-state
			[icon]="IconName.users"
			[smallIcon]="true"
		>
			{{'TEAM__PACKAGE__FALLBACK' | translate}}
		</ui-empty-state>
	</ng-template>
</div>

