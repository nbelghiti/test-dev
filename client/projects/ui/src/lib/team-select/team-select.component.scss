// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Team
// =============================================================================
// .c-team-select {}

// =============================================================================
// :: Elements
// =============================================================================
.c-team-select__item {
	display: flex;
	align-items: flex-start;

	padding: 15rem 0;

	border-top: $border-width-base solid $border-color-grey-76;

	&:first-child {
		border-top: none;
	}
}

.c-team-select__members {
	margin-bottom: 1rem;
}

.c-team-select__member {
	display: flex;
	align-items: center;

	margin-top: 1rem;
}

// =============================================================================
// :: modifiers
// =============================================================================
.c-team-select--first {
	padding: 1rem;

	background-color: $orange-94;
}
