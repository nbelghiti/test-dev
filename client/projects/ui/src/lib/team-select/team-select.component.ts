import { Component, ContentChildren, EventEmitter, Input, Output, QueryList } from '@angular/core';

import { IconName } from '../icon/icon.types';
import { ListItemDirective } from '../list/list-item.directive';

@Component({
	selector: 'ui-team-select',
	templateUrl: './team-select.component.html',
	styleUrls: ['./team-select.component.scss']
})

export class TeamSelectComponent {
	@ContentChildren(ListItemDirective) members: QueryList<ListItemDirective>;
	@Input() public title: string;
	@Input() public action: boolean = true;
	@Output() public newMember: EventEmitter<Event> = new EventEmitter<Event>();
	@Input() public first: boolean = false;

	public IconName: typeof IconName = IconName;
}
