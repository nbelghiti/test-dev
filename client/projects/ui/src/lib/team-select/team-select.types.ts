export interface ISelectedTeamMember {
	limit: number;
	packageId: string;
	type: string;
}

export interface ISelectedTeam<T = ISelectedTeamMember> {
	members: T[];
	title: string;
	totalLimit: number;
}
