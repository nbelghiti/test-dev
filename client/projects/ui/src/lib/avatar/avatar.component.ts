import { Component, Input } from '@angular/core';

import { getInitials } from './avatar.utils';

@Component({
	selector: 'ui-avatar',
	templateUrl: './avatar.component.html',
	styleUrls: ['./avatar.component.scss']
})

export class AvatarComponent {
	@Input() public compact: boolean = false;

	public initials = '';

	@Input() set name(name: string) {
		if (name) {
			this.initials = getInitials(name);
		}
	}
}
