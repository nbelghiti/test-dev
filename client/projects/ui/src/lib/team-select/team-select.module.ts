import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { TranslateModule } from '@ngx-translate/core';

import { EmptyStateModule } from '../empty-state/empty-state.module';
import { HeadingModule } from '../heading/heading.module';
import { LinkModule } from '../link/link.module';
import { UserModule } from '../user/user.module';

import { TeamSelectComponent } from './team-select.component';

@NgModule({
	imports: [
		CommonModule,
		UserModule,
		HeadingModule,
		LinkModule,
		EmptyStateModule,
		RouterModule,
		TranslateModule
	],
	declarations: [
		TeamSelectComponent,
	],
	exports: [
		TeamSelectComponent,
	],
})
export class TeamSelectModule {}
