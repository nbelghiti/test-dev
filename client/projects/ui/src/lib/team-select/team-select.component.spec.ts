import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { TranslateModule } from '@ngx-translate/core';

import { AvatarComponent } from '../avatar/avatar.component';
import { EmptyStateComponent } from '../empty-state/empty-state.component';
import { HeadingComponent } from '../heading/heading.component';
import { IconComponent } from '../icon/icon.component';
import { LinkComponent } from '../link/link.component';
import { UserComponent } from '../user/user.component';

import { TeamSelectComponent } from './team-select.component';

describe('TeamSelectComponent', () => {
	let component: TeamSelectComponent;
	let fixture: ComponentFixture<TeamSelectComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				RouterTestingModule,
				TranslateModule.forRoot()
			],
			declarations: [
				UserComponent,
				AvatarComponent,
				HeadingComponent,
				IconComponent,
				LinkComponent,
				EmptyStateComponent,
				TeamSelectComponent
			]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(TeamSelectComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
