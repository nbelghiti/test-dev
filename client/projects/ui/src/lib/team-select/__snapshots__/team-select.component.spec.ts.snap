// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamSelectComponent Should render correctly 1`] = `
<ui-team-select
  IconName={[Function Object]}
  action={[Function Boolean]}
  first="false"
  members={[Function QueryList]}
  newMember={[Function EventEmitter]}
>
  <div
    class="c-team-select"
  >
    <ui-heading
      ng-reflect-level="3"
    >
      <div
        class="c-heading"
        ng-reflect-klass="c-heading "
        ng-reflect-ng-class="[object Object]"
      >
        
        
        
        
        <h3>
          
        </h3>
        <div
          class="c-heading__meta"
        >
          
          
          
        </div>
      </div>
    </ui-heading>
    <div>
      
      <ui-empty-state
        ng-reflect-icon="users"
        ng-reflect-small-icon="true"
      >
        <div
          class="c-empty-state"
          ng-reflect-klass="c-empty-state"
          ng-reflect-ng-class="[object Object]"
        >
          <span
            class="c-empty-state__icon c-empty-state__icon--small"
            ng-reflect-klass="c-empty-state__icon"
            ng-reflect-ng-class="[object Object]"
          >
            <i
              class="fas fa-users"
            />
          </span>
          <p
            class="c-empty-state__message"
          >
            
          </p>
          <div
            class="c-empty-state__sub-message"
            ng-reflect-ng-class="[object Object]"
          >
             TEAM__PACKAGE__FALLBACK 
          </div>
        </div>
      </ui-empty-state>
      
      <ui-link
        iconbefore="plus"
        label="New team member"
        ng-reflect-icon-before="plus"
        ng-reflect-label="New team member"
      >
        
        <button
          class="c-link "
          type="button"
        >
          
          <div
            class="c-link__icon c-link__icon--before"
          >
            <ui-icon
              class="c-icon"
              ng-reflect-name="plus"
            >
              <i
                class="c-icon__glyph   fas fa-plus"
              />
            </ui-icon>
          </div>
          <span
            class="c-link__text"
          >
            New team member
          </span>
          
        </button>
        
      </ui-link>
    </div>
    
  </div>
</ui-team-select>
`;
